### PR TITLE
Add annotation and text layer factory interfaces for PDFPageView

### PR DESCRIPTION
**Pull request by seanburke-wf**
These are optional parameters to the PDFPageView which allow building UI around document annotations and rendering of text for selection, copy/paste, etc.

Add the necessary interfaces to allow dart users to implement these, or to pass in the default factories provided by PDF.js.

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Module Findings
No security relevant content detected. Please review for security relevance and request security review as needed.

File: lib/src/interfaces.dart [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+part of pdfjs;
+
+class AnnotationLayerBuilderOptions {
+  JsObject _jsInternal;
+  IL10n _l10n;
+  PDFPageProxy _pdfPage;
+
+  AnnotationLayerBuilderOptions() {
+    _jsInternal = new JsObject.jsify({});
+  }
```
Comment (tomconnell-wf) [inline lib/src/interfaces.dart]: You could have the options be params in the constructor, but 🤷‍♂️

**Comment by tomconnell-wf** [comment]
# +1
It looks like the api I expected we would need to implement.  
tridentimp_hellfire_by_shadowhawkdragon-dbc2vff

**Comment by tomconnell-wf** [comment]
# Security +1
The Aviary warning is a red herring on the update of 2017 -> 2018

**Comment by tomconnell-wf** [comment]
# +10 
Via https://github.com/Workiva/w_viewer/pull/463

**Comment by tomconnell-wf** [comment]
# Quality Review Approval: QA +1  
- [x] Testing Instructions
- [x] Dev +1's
- [x] Dev/QA +10 detailing what was tested
- [ ] Tests created, updated, or explained if not present
- [x] All CI checks pass (Smithy, Shipyard, Skynet)
@Workiva/release-management-pp - Rosie, will you automerge?

**Pull request merged**